src/core_modules/capture-core/components/Pages/ViewEvent/epics/viewEvent.epics.js

```diff
@@ -134,18 +137,22 @@ export const backToMainPageEpic = (action$: InputObservable, store: ReduxStore)
             return noWorkingListUpdateNeededOnBackToMainPage();
         }));
 
-export const backToMainPageLocationChangeEpic = (action$: InputObservable, store: ReduxStore) =>
+export const backToMainPageLocationChangeEpic = (action$: InputObservable, store: ReduxStore, { history }: ApiUtils) =>
     action$.pipe(
         ofType(viewEventActionTypes.START_GO_BACK_TO_MAIN_PAGE),
         map(() => {
+            // TODO - This should probably be replaced by URL args
             const state = store.value;
             const programId = state.currentSelections.programId;
             const orgUnitId = state.currentSelections.orgUnitId;
             const showaccessible = state.currentSelections.showaccessible;
+            // const { programId, orgUnitId, showaccessible } = deriveURLParamsFromHistory(history);
```

Comment:
Hey,
Is there any reason why the ids cannot be currently replaced by URL args?
Thanks!

Reply:
Hey,
This is because it's used in the viewEvent-page. The viewEvent-url does not follow the conventional standard of having the id's in the URL, instead it's just `viewEventId=`. This should probably be rewritten if we want to move away from currentSelections. (although in a different issue)
Should I create one?
Thanks.

Reply:
This makes sense, thank you for the explanation!
There is no need to create another ticket, when we will remove the `currentSelections` logic this refactor will come organically.